invenio/base/factory.py

```diff
@@ -155,6 +152,12 @@ def configure_warnings():
         warnings.simplefilter("default", DeprecationWarning)
 
 
+def configure_deprecation_warnings(app):
+    """Configure deprecation warnings if `DEBUG=True` or ."""
+    if not app.debug and not app.config.get("DEPRECATION_WARNINGS_PRODUCTION_ENABLED"):
```

Comment:
Can we move the condition before `warnings.simplefilter("default", DeprecationWarning)`?